### ISSUE TITLE
Try to improve a11y of our dashboard

### ISSUE THREAD

**Issue by martinthomson** [mozilla/standards-positions#980]
Without a label on the left-hand column, it's not *completely* obvious that the expand button operates the way it does.  Also, the fact that we add another column is surprising when you are using a screen reader - the new content tends not to be presented automatically as a result.

We could experiment with `<details>` here, but the fact that we have a table makes that somewhat difficult to get visual styling right.

**Comment by zcorpan** [comment]
> Also, the fact that we add another column is surprising when you are using a screen reader - the new content tends not to be presented automatically as a result.

Another row, right?

I wonder if "Description" and "Mozilla's Position" should instead be in columns of the table. That might make it more natural to navigate with a screen reader. The expand/collapse is there to make it easier to scan the table visually, but with a screen reader you can navigate through a column anyway and the expand/collapse seems like additional burden.

**Comment by zcorpan** [comment]
Mockup:

<img width="932" alt="Screenshot of the table, with the headers Specification, Description, Mozilla Position, More Info. The Detailed Description is in the Description column, and Mozilla's Position text is in the Mozilla Position column, after the 'positive' button." src="https://github.com/mozilla/standards-positions/assets/244772/6585191e-6118-4b40-a980-846182e0f62c">

The disclosure triangle could visually hide the description and detailed position, but still have it exposed for ATs.

**Comment by martinthomson** [comment]
The narrow column really hurts readability.  Is there some way to have the descriptive text in the content model as a separate column, but use grid layout placement to have it appear on the next row?  I've had some success with fairly adventurous CSS grid placement strategies like that in the past.

**Comment by jcsteh** [comment]
Visually hiding something that is shown for AT isn't great, particularly for users using multiple modalities; e.g. a partially sighted user that uses both magnification and speech.

Another (slightly hacky) possibility: keep things as they are, but:

1. Give the description column a header (so it's more obvious that this column will contain the description). I think that should be done regardless of what else is done here.
2. When the expand link is pressed, focus the text of the description. That will at least avoid a screen reader user not knowing where to find the description that just appeared. To do that, the description will need to be in a span or similar to avoid screen readers switching out of document browsing mode. Example:
  ```
  <table>
    <tr><th>Description</th></tr>
    <tr><td><a onclick="descRow.hidden = false; descText.focus();">Expand</a></td></tr>
    <tr id="descRow" hidden><td><span id="descText" tabindex="0">Some long description</span></td></tr>
  </table>
  ```
  This certainly isn't best practice, but it might balance the visual and a11y concerns here.

**Comment by zcorpan** [comment]
Hmm. Maybe we can put the description and detailed description text in the same cell as the disclosure triangle, but make it appear visually on its own row (like @martinthomson suggested). Then we don't need to make the text focusable (which it nominally shouldn't be) or move focus around, which might be disorienting. We can even use the `details` element directly.

**Comment by martinthomson** [comment]
Not a bad idea.  Ideally we'd play with tabindex for the ordering of columns so that the details came *after* the title.

**Comment by martinthomson** [comment]
I hacked on something in a demo, see https://developer.mozilla.org/en-US/play?id=NczmHqWDTDwjiEHnVSHQ6YcgSLmAEzaL0mpy%2FU5GQMMclX5miLIpaN0%2FOa4GfQ8kk8oT82ye1w6egDis

I couldn't work out why the subgrid wasn't placing `<summary>` where I asked it to, but you can see how this might work.

**Comment by martinthomson** [comment]
According to @dholbert, this is down to `<details>` always being `display: block`, even if you set it to something else.  That's an open issue on the HTML spec, but it might take some time to get that resolved.  In the meaning, it should be easy to use `<div class="summary">` or similar and style with subgrid as in the linked demo.  If I get some time, I'll give that a swing.

**Comment by zcorpan** [comment]
This should be fixed by #1063. The `details` is on a separate row, but visually it looks like the `summary` is in the first column of the prior row.
